### PR TITLE
Issue #2 React Router Dom

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delineated between viewing and adding.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* `react-router-dom` has been added as a project dependency
* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text

Notes:

* Some pairs have gone straight to wanting to integrate a CSS framework to achieve the “links pinned to the bottom of the page” AC here. You're encouraged to keep it simple at this point. Perhaps look into `position: fixed;` or perhaps look into a tutorial on CSS grid. There will be time to get fancy later in the project.

### PR DESCRIPTION
**Pull request by k-schrattenholzer**
adds react router dom dependency. adds home, list and add item views, with navigation links in the footer that bold when active

## Description

This PR adds a navigation Footer, with links to Add Item, List and Home views using React Router Dom.

We were unsure if a Home view was necessary, but when we were declaring the App routes, decided to add one as the rendered view for the '/' landing path.

## Related Issue

closes #2 

## Acceptance Criteria

1. react-router-dom has been added as a project dependency
2. Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
3. When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
4. Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

image
image

## Testing Steps / QA Criteria

Click links, check url, and active page link should be bolded.

### CODE REVIEW

File: src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { NavLink } from 'react-router-dom';
+import './Footer.css';
+
+export default function Footer() {
+  return (
+    <div className="footer">
+      <NavLink to="/" activeClassName="active" className="navLinks">
```
Comment (lindseyindev) [inline src/components/Footer/Footer.jsx]: Nice use of activeClassName! I hadn't seen that property before so I went to the react-router docs and the strange thing is the docs say the property was removed from v6 but it clearly works in this feature using React Router v6.3

File: src/views/Home/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function Home() {
```
Comment (lindseyindev) [inline src/views/Home/Home.jsx:1]: Great job including this route and adding these components!

**Review by artsycoder533 — APPROVED** [review]
Good job guys!  The all the links navigate to the corresponding view and the active view is indicated in bold.  The footer is also visible on every view.  As a heads up EJ pointed out that in the footer component activeClassName="active" is not needed.  We tried removing this and the styling of the active link still works.  The NavLink has a built in active class that styles the active link as long as you give it special styling!  Either way it still works!

**Review by mxmason — APPROVED** [review]
Great work, everyone! I'm not sure how this branch's `package-lock.json` hit a state of conflict, but it happens sometimes. We can resolve that during office hours, if necessary! Let us know if you need help!

File: src/components/Footer/Footer.jsx (hunk 1 above)
Comment (mxmason) [reply src/components/Footer/Footer.jsx]: Huh. TIL that `activeClassName` was removed from React Router. cc @sparsell, @artsycoder533 – this is relevant to our discussion!

**Comment by k-schrattenholzer** [comment]
Cool! I like how that activeClassName is baked in. One of those v6 features I haven't run into yet

**Review by sparsell — APPROVED** [review]
Updated dependency to install react-router. 
All links appeared as expected. 
URLs corresponded with links and were bold when on active page. 

All checks passed. Great job, all!

**Pull request merged**